lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1012,59 +1029,107 @@ public function orderBy(string $sort, ?string $order = null)
      */
     public function addOrderBy(string $sort, ?string $order = null)
     {
-        return $this->add('orderBy', $sort . ' ' . (! $order ? 'ASC' : $order), true);
+        $this->queryParts->orderBy[] = $sort . ' ' . (! $order ? 'ASC' : $order);
+
+        $this->state = self::STATE_DIRTY;
+
+        return $this;
     }
 
     /**
-     * Gets a query part by its name.
+     * Gets all query parts.
      *
-     * @return mixed
+     * @internal For tests only.
      */
-    public function getQueryPart(string $queryPartName)
+    public function getQueryParts() : QueryParts
```

Comment:
**Should we keep this method?** It serves little purpose, it's actually only used in [this test](https://github.com/doctrine/dbal/blob/dbd94c9d591e3d6f980f3a077cfa5b64c600b6d7/tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php#L733).
I've marked it as `@internal` for now.

Reply:
It's an implementation detail. I'd rather remove it completely. The end result of using the builder is the query, not the parts.

Reply:
OK to remove [this test](https://github.com/doctrine/dbal/blob/dbd94c9d591e3d6f980f3a077cfa5b64c600b6d7/tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php#L733) then?

Reply:
Yes.

Reply:
Done in 682b5bb364ddf26d85f16bf9cf63f1565aeb0c42.